python/ray/dashboard/modules/job/common.py

```diff
@@ -211,8 +236,54 @@ async def put_info(
             overwrite,
             namespace=ray_constants.KV_NAMESPACE_JOB,
         )
+        if added_num == 1 or overwrite:
+            # Write export event if data was updated in the KV store
+            try:
+                self._write_submission_job_export_event(job_id, job_info)
+            except Exception:
+                logger.exception("Error while writing job submission export event.")
         return added_num == 1
 
+    def _write_submission_job_export_event(
+        self, job_id: str, job_info: JobInfo
+    ) -> None:
+        """
+        Write Submission Job export event if _export_submission_job_event_logger
+        exists. The logger will exist if the export API feature flag is enabled
+        and a log directory was passed to JobInfoStorageClient.
+        """
+        if not self._export_submission_job_event_logger:
+            return
+
+        status_value_descriptor = (
+            ExportSubmissionJobEventData.JobStatus.DESCRIPTOR.values_by_name.get(
+                job_info.status.name
+            )
+        )
+        if status_value_descriptor is None:
```

Comment:
when does it happen?

Reply:
When a new JobInfo status is added to the python model without updating the proto. We do have a comment to keep these in sync though, so this should be unlikely.